File: Test/DemoTemplate.cpp
#include "Application.h"
#include "MainWindow.h"

using namespace d14uikit;

#define DEMO_NAME L"DemoTemplate"

int main(int argc, char* argv[])
{
    float dpi = 96.0f;
    if (argc >= 2 && strcmp(argv[1], "HighDPI"))
    {
        dpi = 192.0f;
    }
    Application app(DEMO_NAME, dpi);

    //------------------------------------------- Initialize UI objects.

    MainWindow mwnd;

    //------------------------------------------- Set UI event callacks.

    return app.run();
}

File: Test/PyBind/DemoTemplate.py
from sys import argv

from D14UIKit import *

DEMO_NAME = 'DemoTemplate'

if __name__ == '__main__':
    dpi = 96.0
    if len(argv) >= 2 and argv[1] == 'HighDPI':
        dpi = 192.0

    app = Application(DEMO_NAME, dpi)

    #------------------------------------------- Initialize UI objects.

    mwnd = MainWindow(DEMO_NAME)

    #------------------------------------------- Set UI event callacks.

    exit(app.run())
